src/pam_google_authenticator.c

```diff
@@ -285,7 +285,6 @@ static char *get_secret_filename(pam_handle_t *pamh, const Params *params,
         log_message(LOG_ERR, pamh, "Short mem allocation failed");
         goto errout;
       }
-      var += resized - secret_filename;
```

Comment:
No, that's not right. This change will actually *break* the `realloc` logic.

`var` is an offset into `secret_filename`, and this line adjust it to be an offset into `resized`.

Though by the C standard it does seem to actually be undefined behaviour:

>  J.2.  The behavior is undefined in the following circumstances: […] The value of a pointer that refers to space deallocated by a call to the free or realloc function is used (7.22.3)

I suppose this "uses" the pointers. So before the realloc `var` should be turned into an offset int, and turned back into a pointer afterwards.

In C++ it seems like it's implementation-defined:

> If the argument given to a deallocation function in the standard
> library is a pointer that is not the null pointer value (4.10), the
> deallocation function shall deallocate the storage referenced by the
> pointer, rendering invalid all pointers referring to any part of the
> deallocated storage. Indirection through an invalid pointer value and
> passing an invalid pointer value to a deallocation function have
> undefined behavior. Any other use of an invalid pointer value has
> implementation-defined behavior

Reply:
Fixed in https://github.com/google/google-authenticator-libpam/commit/59d1cc4f5dfff75efb0b541f7c2a2d25f986ad68